fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LucenOnlineIndexingTest.java

```diff
@@ -0,0 +1,221 @@
+        FDBDatabaseFactory.instance().setMaxDelayMillis(oldMaxDelayMillis);
+        FDBDatabaseFactory.instance().setInitialDelayMillis(oldInitialDelayMillis);
+        FDBDatabaseFactory.instance().setMaxAttempts(oldMaxAttempts);
+    }
+
+    @BeforeEach
+    public void setUp() {
+        fdb.run(context -> {
+            FDBRecordStore.deleteStore(context, path);
+            return null;
+        });
+    }
+
+    private void rebuildIndexMetaData(final FDBRecordContext context, final String document, final Index index) {
+        Pair<FDBRecordStore, QueryPlanner> pair = LuceneIndexTestUtils.rebuildIndexMetaData(context, path, document, index, useRewritePlanner);
+        this.recordStore = pair.getLeft();
+        this.planner = pair.getRight();
+    }
+
+    private void disableIndex(Index index, String document) {
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, document, index);
+            recordStore.markIndexDisabled(index).join();
+            context.commit();
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void luceneOnlineIndexingTestSimple(boolean preventMergeDuringIndexing) {
+        Index index = SIMPLE_TEXT_SUFFIXES;
+        disableIndex(index, SIMPLE_DOC);
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.saveRecord(createSimpleDocument(1623L, ENGINEER_JOKE, 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, WAYLON, 1));
+            recordStore.saveRecord(createSimpleDocument(2222L, WAYLON + " who?", 1));
+            context.commit();
+        }
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.saveRecord(createSimpleDocument(1623L, ENGINEER_JOKE, 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, WAYLON, 1));
+            recordStore.saveRecord(createSimpleDocument(2222L, WAYLON + " who?", 1));
+            context.commit();
+        }
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                    .setRecordStore(recordStore)
+                    .setIndex(index)
+                    .setIndexingPolicy(OnlineIndexer.IndexingPolicy.newBuilder()
+                            .setPreventLuceneMergeDuringIndexing(preventMergeDuringIndexing))
+                    .build()) {
+                assertTrue(recordStore.isIndexDisabled(index));
+                indexBuilder.buildIndex(true);
+            }
+        }
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            assertTrue(recordStore.isIndexReadable(index));
```

Comment:
It involves accessing internal bits, but it seems like this should assert that there is a reasonable number of segments.

Reply:
Added assertion for `allFiles.length < 10`.

Reply:
You could try commenting out the code that actually does the merges, and I think it might end up with more than 10 segments, especially if it only indexes one record per transaction.

Reply:
✅